### PR TITLE
Upgrade to Babel 7

### PR DESCRIPTION
**Pull request by keeganwitt**
### What

Upgrade to Babel 7.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -35,16 +35,17 @@
   "license": "MIT",
   "repository": "jest-community/jest-extended",
   "devDependencies": {
+    "@babel/cli": "^7.0.0",
+    "@babel/core": "^7.0.0",
+    "@babel/plugin-proposal-object-rest-spread": "^7.0.0",
+    "@babel/plugin-transform-async-to-generator": "^7.0.0",
+    "@babel/plugin-transform-modules-commonjs": "^7.0.0",
```
Comment (SimenB) [inline package.json:42]: we don't need any of these when we use the `env` preset

File: package.json [hunk 2]
```diff
@@ -95,11 +96,21 @@
     ]
   },
   "babel": {
+    "presets": [
+      [
+        "@babel/preset-env",
+        {
+          "targets": {
+            "node": "12"
```
Comment (SimenB) [inline package.json:104]: this should target whatever is the lowest this library supports - node 8 from the looks of https://github.com/jest-community/jest-extended/blob/437d55ae2e5e3226c9c3cd8bc2317965f9a52a8f/.travis.yml#L3

Comment (keeganwitt) [reply package.json:104]: Is that still the case? I think that file hasn't been updated in a while. What does Jest support?

File: package.json (hunk 1 above)
Comment (keeganwitt) [reply package.json:42]: Are they still needed to be mentioned in the plugins list? Or does that come through present env also? I wasn't totally certain.

```json
"plugins": [
  "@babel/plugin-transform-modules-commonjs",
  "@babel/plugin-proposal-object-rest-spread",
  "@babel/plugin-transform-async-to-generator",
  "module:babel-jest-assertions",
  "gwt",
  "macros"
]
```

Comment (SimenB) [reply package.json:42]: Remove all mention of them, they're included in the env preset 🙂

Comment (keeganwitt) [reply package.json]: Awesome. Thanks for confirming. Removed.

File: package.json (hunk 2 above)
Comment (SimenB) [reply package.json:101]: Jest 26 and 27 supports node 10, but I'm fine with only supporting 12 and later here as 10 is EOL

**Comment by keeganwitt** [comment]
I suspect the failures are because the current version of ESLint doesn't play nice with the upgraded version of `@babel/eslint-parser`. Once we merge #328, I'll rebase and look again at these failures.

**Comment by keeganwitt** [comment]
I went ahead and added the lint fixes to this branch so it can be merged independently.

**Pull request merged**